### PR TITLE
feat(solana): add support for. token account info

### PR DESCRIPTION
**Pull request by gabrielKerekes**
Allow passing token accounts info into Trezor so that Trezor could verifiably display a token account's base address instead of the token address.

Also Suite will now use only associated token accounts for recipient accounts.

_The PR might change significantly depending on the feedback on firmware PR https://github.com/vacuumlabs/trezor-firmware/pull/87._

### CODE REVIEW

File: packages/connect/src/api/solana/additionalInfo.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
+import { SolanaTxAdditionalInfo } from '../../types/api/solana';
+import { validateParams } from '../common/paramsValidator';
+
+const validateAdditionalInfo = (additionalInfo: SolanaTxAdditionalInfo) => {
+    validateParams(additionalInfo, [{ name: 'tokenAccountsInfos', type: 'array' }]);
+
+    additionalInfo.tokenAccountsInfos?.forEach(tokenAccountInfo => {
+        validateParams(tokenAccountInfo, [
+            { name: 'baseAddress', type: 'string' },
```
Comment (marekrjpolak) [inline packages/connect/src/api/solana/additionalInfo.ts]: Nit: without `required: true`, omitting is allowed, so decide whether it's what you want or not.

File: packages/suite/src/utils/wallet/solanaUtils.ts [hunk 2]
```diff
@@ -216,9 +233,9 @@ export const buildTokenTransferTransaction = async (
     let finalReceiverAddress = toAddress;
     if (isReceiverAddressSystemAccount) {
         // Step 3: If not, check if the receiver owns an associated token account
-        if (toTokenAccounts && toTokenAccounts.length > 0) {
+        if (toTokenAccount != null) {
```
Comment (marekrjpolak) [inline packages/suite/src/utils/wallet/solanaUtils.ts]: We're not using loose equality operators, also there should never be `null` in this case.
```suggestion
        if (toTokenAccount) {
```

**Review by marekrjpolak — COMMENTED** [review]
Comment for future ourselves: We should generate protobuf messages properly (from `trezor-common`) after everything is merged in fw repo.

File: packages/connect/src/api/solana/additionalInfo.ts (hunk 1 above)
Comment (gabrielKerekes) [reply packages/connect/src/api/solana/additionalInfo.ts]: Aah, nice, thanks, it should indeed be required. Updated. Sorry, I had to do a force push :/  https://github.com/trezor/trezor-suite/pull/10164/commits/725053316abca589e4658c268a19a7d8e750c411

File: packages/suite/src/utils/wallet/solanaUtils.ts (hunk 2 above)
Comment (gabrielKerekes) [reply packages/suite/src/utils/wallet/solanaUtils.ts]: Updated in https://github.com/trezor/trezor-suite/pull/10164/commits/921b7204a47adeb817fba4d818ca5ce87e2f1ba0#diff-143383856d0b16d79c6ef3b53799bfd67f1b72d5558060a3ca90fdc262ad5982R230. (again, sorry for the force push).

**Comment by gabrielKerekes** [comment]
I had to forcepush because I was fixing failing tests before there was any review and apparently I forgot to force push before a review came in. But the GH "Compare" functionality shows the diff nicely 👍

**Pull request merged**